LandSandBoat/server
# [core] [sql] Update Effect Flags, Comments, Job Change Behavior

**_I affirm:_**

## What does this pull request do?

Updates EFFECTFLAG_ comments
Updates Ability Effect Flags to add `EFFECTFLAG_ON_JOBCHANGE` to relevant effects 
Updates Job Change Packet to also explicitly clear `EFFECTFLAG_ROLL` as there are no negative Corsair Rolls (Bust is not flagged as a roll).

Closes #2715 

**Changes based on Retail observations (Using Nomad Moogle in Selbina):**

* 1HR ability effects wears off on job change (Perfect Dodge drops when changing THF -> WAR) 
* Corsair rolls wear on job change (Corsair's Roll drops when changing COR -> THF) 
* Bard songs wear on job change (elegy not observed/tested) 
* Quickening Wears on job change (Use Sprinter's Shoes -> Change Job) 
* Meditate has no effect and TP Gain does NOT drop on job change (Use Mediate, Change SAM -> WAR, TP Still increases)

## Steps to test these changes

Zone into Selbina
`!additem 15754`
`!changejob SCH 99`
Equip and use Sprinters Shoes
Use Light Arts
Change jobs using Nomad Moogle
Observe both effects drop.
`!changejob COR 99`
`!additem 5493`
Use Corsair's Die
Use Corsair's Roll
Change jobs using the Nomad Moogle
Observe Corsair's Roll and Double Up Chance effects drop.
